packages/nimble-components/src/wafer-map/specs/features/change-highlight.md

```diff
@@ -0,0 +1,56 @@
+# Nimble Wafer Map Die Highlighting
+
+## Problem Statement
+
+The wafer map receives an array of strings as the values that will be highlighted.
+
+```
+@observable public highlightedValues: string[] = [];
+```
+
+In the following case the highlighted values are 2 and 3.
+
+![Highlighted Values](resources/highlighted-values.png)
+
+The current approach is limited. We can either highlight all the dies having 2s and 3s or neither of them. The customer needs exceeds this approach. The customer may see an anomaly on the dies having DieX: 15 and Soft Bin: 2 and wishes to highlight all these values, but the current code is not capable of this.
+
+## Links To Relevant Work Items and Reference Material
+
+N\A
+
+## Implementation / Design
+
+We will replace the old logic, removing it completely, and add a new one.
+
+Each die will have an optional field called "tags", being optional in the cases when we don't use highlighting we can get rid of it.
+
+```
+export interface WaferMapDie{
+...
+   tags?: string[];
+...
+}
+```
+
+Example of tags:
+
+```
+{ 'sb1', 'hb2' }
+```
+
+We will make the highlights based on this tags. The wafer will receive a list of highlightedTags
+
+```
+@observable public highlightedTags: string[][]
+```
+
+Example:
+
+```
+[
+    [{ 'sb2', 'hb3' }],
+    [{ 'sb1' }]
+]
```

Comment:
The spec should be precise and consider edge cases:
- This example is not valid syntax. It shows an array of arrays of objects. Provide realistic examples. `[['sb2', 'hb3'], ['sb1']]`
- Is this level of complexity actually needed? Can it just be a list of tags where if you match any tag in the list we highlight? i.e. ['a', 'b'] matches 'a' or 'b'. If this is not enough can you capture it in an alternatives section and describe why it is insufficient?
- If the complexity shown is needed, can you provide some clear use cases of what you are considering?
  - The underlying question is why are these rules enough? Do you need to be able to say something like "must have tag a and b but not c"? Should we have an entire Set Theory engine (intersection, union, complement, difference, etc)? Should we have an entire selection engine like CSS `.a.b:not(.c)`? Do we need to make it even more extensible, like you can plugin completely custom logic and if so what are the specific use cases?
- Clearly describe the rules and validation states. This is groups of tags. So describe precisely, examples:
    - If a wafer matches all the tags in a group then it is highlights. It must match all the tags in at least one group to highlight.
    - What happens if a group is an empty array? Is it ignored? Does it match everything?
    - What happens if a tag does not actually exist in the dies set? Is it ignored?
- Include a section about significant alternatives that we have discussed. Definitely why a single list of tags is insufficient if that's the case. But also some of the other patterns.

Reply:
Hi Milan, I tried to address all your topics inside the spec, you can take a look